### PR TITLE
Make data upload date columns case insensitive

### PR DESCRIPTION
**Pull request by ericboucher**
Make data upload date columns case insensitive and other improevements

### CODE REVIEW

File: packages/api/src/utils/uploads/upload-sheet-data.ts [hunk 1]
```diff
@@ -73,12 +73,27 @@ const IGNORE_HEADER_KEYS = [
   'Site Name',
   'Time (HH:mm:ss)',
   'Time (Fract. Sec)',
-];
+].map((key) => key.toLowerCase());
```
Comment (ericboucher) [inline packages/api/src/utils/uploads/upload-sheet-data.ts:76]: @tleraj is there a smarter way to do this?

File: packages/api/src/utils/uploads/upload-sheet-data.ts [hunk 2]
```diff
@@ -190,6 +212,7 @@ const getTimestampFromDateString = (dataObject: any) => {
   );
 
   if (!dateKey) {
+    // TODO - Can we raise an error here instead? Is it safe?
```
Comment (ericboucher) [inline packages/api/src/utils/uploads/upload-sheet-data.ts:215]: @tleraj wdyt?

File: packages/api/src/utils/uploads/upload-sheet-data.ts [hunk 3]
```diff
@@ -445,6 +468,7 @@ export const uploadTimeSeriesData = async (
       'timestamp',
     );
 
+    // TODO - Why do we have this check on surveyPoint?
```
Comment (ericboucher) [inline packages/api/src/utils/uploads/upload-sheet-data.ts:471]: @tleraj can you explain why we have this check? And document if needed?

File: packages/api/src/utils/uploads/upload-sheet-data.ts (hunk 1 above)
Comment (tleraj) [reply packages/api/src/utils/uploads/upload-sheet-data.ts:76]: Not sure what you mean. We can store the array as lowercase in the first place, i.e. 

```typescript
const IGNORE_HEADER_KEYS = [
  '#',
  'site name',
  'time (hh:mm:ss)',
  'time (fract. sec)'
];
```

File: packages/api/src/utils/uploads/upload-sheet-data.ts (hunk 2 above)
Comment (tleraj) [reply packages/api/src/utils/uploads/upload-sheet-data.ts:215]: Yeah, sounds good!! :+1:

File: packages/api/src/utils/uploads/upload-sheet-data.ts (hunk 1 above)
Comment (tleraj) [reply packages/api/src/utils/uploads/upload-sheet-data.ts:76]: Not sure what you mean. We could store this array to lowercase in the first place, i.e.
```typescript
const IGNORE_HEADER_KEYS = [
  '#',
  'site name',
  'time (hh:mm:ss)',
  'time (fract. sec)'
]
```

File: packages/api/src/utils/uploads/upload-sheet-data.ts (hunk 3 above)
Comment (tleraj) [reply packages/api/src/utils/uploads/upload-sheet-data.ts:471]: On the frontend, we need to specify a survey point when we upload sonde or metlog data. If this is not necessary, we can remove this if statement.

Comment (ericboucher) [reply packages/api/src/utils/uploads/upload-sheet-data.ts:471]: I think it should be necessary on the frontend for now, but not on the backend

**Pull request merged**